lib/phonegap/cordova-dependence.js

```diff
@@ -69,9 +72,25 @@ module.exports.exec = function(projectDir, extEvents) {
                 var cordovaCommand = 'npm install cordova --save';
                 extEvents.emit('log', 'Adding Cordova dependency with: "' + cordovaCommand + '"');
                 extEvents.emit('log', 'This may take a few minutes ...');
+
+                var spinner = ora('installing cordova').start();
+
+                var startTime = Date.now();
+                analytics.trackEvent([], 0, {
+                    short_message: "npm-install-cordova-start",
+                    _params: []
+                });
+
                 var child = shell.exec(cordovaCommand, {
                     silent: true
                 }, function(code, stdout, stderr) {
+                    spinner.stop();
+                    var dT = Date.now() - startTime;
+                    var dTsec = Math.round(dT / 1000) + "s"; // analytics does not like numbers,
+                    analytics.trackEvent([dTsec], code, {
+                        short_message: "npm-install-cordova-finish",
+                        _params: [dTsec]
```

Comment:
Fortunately the underlying database, ElasticSearch, is a document store, so we don't have to worry about columns/rows. Each document is treated individually and there is no schema, really.

That's the nice thing about ES, is that you have complete freedom about document structure. It keeps set of running indices on all fields for all documents up to date all the time - part of the reason it is so RAM-heavy.